subsys/net/lib/wifi_credentials/wifi_credentials.c

```diff
@@ -132,12 +133,14 @@ int wifi_credentials_get_by_ssid_personal_struct(const char *ssid, size_t ssid_l
 		goto exit;
 	}
 
-	if (buf->header.type != WIFI_SECURITY_TYPE_NONE &&
-	    buf->header.type != WIFI_SECURITY_TYPE_PSK &&
-	    buf->header.type != WIFI_SECURITY_TYPE_PSK_SHA256 &&
-	    buf->header.type != WIFI_SECURITY_TYPE_SAE &&
-	    buf->header.type != WIFI_SECURITY_TYPE_EAP_TLS &&
-	    buf->header.type != WIFI_SECURITY_TYPE_WPA_PSK) {
+	for (int security_type = 0; security_type <= WIFI_SECURITY_TYPE_MAX; security_type++) {
+		if (buf->header.type == security_type) {
```

Comment:
Can't we just check if `buf->header.type < 0 || buf->header.type > WIFI_SECURITY_TYPE_MAX` instead of looping all of the integers in range?

Reply:
Updated.